src/com/magento/idea/magento2plugin/actions/generation/dialog/NewWebApiDeclarationDialog.java

```diff
@@ -0,0 +1,200 @@
+        this.classFqn = classFqn;
+        this.methodName = methodName;
+
+        setContentPane(contentPane);
+        setModal(true);
+        setTitle(NewWebApiDeclarationAction.ACTION_DESCRIPTION);
+        getRootPane().setDefaultButton(buttonOK);
+
+        buttonOK.addActionListener(event -> onOK());
+        buttonCancel.addActionListener(event -> onCancel());
+
+        // call onCancel() when cross is clicked
+        setDefaultCloseOperation(DO_NOTHING_ON_CLOSE);
+        addWindowListener(new WindowAdapter() {
+            @Override
+            public void windowClosing(final WindowEvent event) {
+                onCancel();
+            }
+        });
+
+        // call onCancel() on ESCAPE
+        contentPane.registerKeyboardAction(
+                event -> onCancel(),
+                KeyStroke.getKeyStroke(KeyEvent.VK_ESCAPE, 0),
+                JComponent.WHEN_ANCESTOR_OF_FOCUSED_COMPONENT
+        );
+
+        fillPredefinedValuesAndDisableInputs();
+    }
+
+    /**
+     * Open New WebApi dialog window.
+     *
+     * @param project Project
+     * @param directory PsiDirectory
+     * @param classFqn String
+     * @param methodName String
+     */
+    public static void open(
+            final @NotNull Project project,
+            final @NotNull PsiDirectory directory,
+            final @NotNull String classFqn,
+            final @NotNull String methodName
+    ) {
+        final NewWebApiDeclarationDialog dialog =
+                new NewWebApiDeclarationDialog(project, directory, classFqn, methodName);
+        dialog.pack();
+        dialog.centerDialog(dialog);
+        dialog.setVisible(true);
+    }
+
+    /**
+     * Fire generation process if all fields are valid.
+     */
+    private void onOK() {
+        if (!validateFormFields()) {
+            return;
+        }
+
+        new WebApiDeclarationGenerator(
+                getDialogDataObject(),
+                project
+        ).generate(NewWebApiDeclarationAction.ACTION_NAME, true);
+
+        this.setVisible(false);
+    }
+
+    /**
+     * Fill predefined values and disable corresponding inputs.
+     */
+    private void fillPredefinedValuesAndDisableInputs() {
+        serviceClass.setText(classFqn);
+        serviceClass.setEditable(false);
+        serviceClass.setEnabled(false);
+        serviceMethod.setText(methodName);
+        serviceMethod.setEditable(false);
+        serviceMethod.setEnabled(false);
+    }
+
+    /**
+     * Create custom components and fill their entries.
+     */
+    @SuppressWarnings({"PMD.UnusedPrivateMethod", "PMD.AvoidInstantiatingObjectsInLoops"})
+    private void createUIComponents() {
+        httpMethod = new ComboBox<>();
+
+        for (final String method : HttpMethod.getHttpMethodList()) {
+            httpMethod.addItem(new ComboBoxItemData(method, method));
+        }
+        aclResource = new FilteredComboBox(new LinkedList<>(Arrays.asList("self", "anonymous")));
```

Comment:
"self", "anonymous" should be moved to an enum

Reply:
Done.